lib/vagrant-cachier/cap/linux/chef_file_cache_path.rb

```diff
@@ -6,7 +6,7 @@ module ChefFileCachePath
           def self.chef_provisioner?(machine)
             provisioners = machine.config.vm.provisioners
             chef_provisioners = [:chef_solo, :chef_client]
-            compat_provisioners = provisioners.keep_if { |p| chef_provisioners.include? p.name }
+            compat_provisioners = provisioners.keep_if { |p| chef_provisioners.include? p.name || p.type }
```

Comment:
@fgrehm @Yserz that might be a bit too optimistic and works only if the provisioner name is not set. However, note that you can have named provisioners in which case the `p.name` might be anything:
https://docs.vagrantup.com/v2/provisioning/basic_usage.html

Here's how I solved it in one of my vagrant plugins:
https://github.com/tknerr/vagrant-toplevel-cookbooks/commit/ae188ee09f572d91efd88c0c071cbd83c44d416d

Or @sethvargo here:
https://github.com/berkshelf/vagrant-berkshelf/blob/cd5ee78f28ee9f6bec1b72fcdbcb042b0155aa04/lib/vagrant-berkshelf/helpers.rb#L69-L86

/cc @cassianoleal